packages/dds/tree/src/shared-tree/defaultCommitEnricher.ts

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils/internal";
+import { ChangeRebaser, GraphCommit, RevisionTag } from "../core/index.js";
+import { ICommitEnricher } from "../shared-tree-core/index.js";
+import { disposeSymbol } from "../util/index.js";
+
+/**
+ * A checkout whose state can be controlled and used to enrich changes with refreshers.
+ */
+export interface ChangeEnricherCheckout<TChange> {
```

Comment:
I acutally picked the "Checkout" part of the name on prupose because the are similar things that play similar roles.
Let me know if the expanded doc comment makes that clearer or if you still think that's too confusing.

Reply:
I see. That makes sense, thanks for updating the doc!